tests/e2e/tests/cart/cart-block.shopper.block_theme.spec.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * External dependencies
+ */
+import { test as base, expect } from '@woocommerce/e2e-playwright-utils';
+import {
+	installPluginFromPHPFile,
+	uninstallPluginFromPHPFile,
+} from '@woocommerce/e2e-mocks/custom-plugins';
+
+/**
+ * Internal dependencies
+ */
+import { CartPage } from './cart.page';
+import {
+	DISCOUNTED_PRODUCT_NAME,
+	REGULAR_PRICED_PRODUCT_NAME,
+} from '../checkout/constants';
+
+const test = base.extend< { pageObject: CartPage } >( {
+	pageObject: async ( { page }, use ) => {
+		const pageObject = new CartPage( {
+			page,
+		} );
+		await use( pageObject );
+	},
+} );
+
+test.describe( 'Shopper → Cart block', () => {
+	test( 'The discount label is only visible next to the discounted product', async ( {
+		pageObject,
+		frontendUtils,
+	} ) => {
+		await frontendUtils.goToShop();
+		await frontendUtils.emptyCart();
+		await frontendUtils.addToCart( DISCOUNTED_PRODUCT_NAME );
+		await frontendUtils.addToCart( REGULAR_PRICED_PRODUCT_NAME );
+		await frontendUtils.goToCart();
+		// Check for discount on the discounted product
+		const discountedProductRow = await pageObject.findProductRow(
+			DISCOUNTED_PRODUCT_NAME
+		);
+		await expect( discountedProductRow ).toBeVisible();
+		const capRegularPriceElement = discountedProductRow?.locator(
+			'.wc-block-components-product-price__regular'
+		);
+		const capDiscountedPriceElement = discountedProductRow?.locator(
+			'.wc-block-components-product-price__value.is-discounted'
+		);
+		await expect( capRegularPriceElement ).toBeVisible();
+		await expect( capDiscountedPriceElement ).toBeVisible();
+
+		// Check for absence of discount on the regular priced product
+		const regularPricedProductRow = await pageObject.findProductRow(
+			REGULAR_PRICED_PRODUCT_NAME
+		);
+		await expect( regularPricedProductRow ).toBeVisible();
+		const hoodieRegularPriceElement = regularPricedProductRow.locator(
+			'.wc-block-components-product-price__regular'
+		);
+		const hoodieDiscountedPriceElement = regularPricedProductRow.locator(
+			'.wc-block-components-product-price__value.is-discounted'
+		);
+		await expect( hoodieRegularPriceElement ).toBeHidden();
+		await expect( hoodieDiscountedPriceElement ).toBeHidden();
+	} );
+
+	test( 'Products with updated prices should not display a discount label', async ( {
+		pageObject,
+		frontendUtils,
+	} ) => {
+		await installPluginFromPHPFile(
+			`${ __dirname }/update-price-plugin.php`
+		);
```

Comment:
Hi, @gigitux! I followed the guidance from [Testing WordPress actions and filters](https://github.com/woocommerce/woocommerce-blocks/blob/7403a5a4c2a23f1f5e94041e2ac1a7dde6dc8805/tests/e2e/mocks/custom-plugins/readme.md#testing-wordpress-actions-and-filters) but didn't achieve the expected outcome. The custom plugin that's supposed to alter product prices in the Cart didn't seem to work; it showed the original prices instead. I tried debugging this issue yesterday without success. Could you help me in identifying any potential steps I might have overlooked?

Reply:
Hi, @tarhi-saad!
I tried to run the test, and the price is updated.

![image](https://github.com/woocommerce/woocommerce-blocks/assets/4463174/3e6d8283-61f6-4648-9fb2-4127d73b9ccc)

I don't think that it is related, but the only change that I made was to add the suffix `.side_effects`. This is necessary because the filter could impact other tests (Mini Cart - for example), so it has to be run NOT in parallel.